ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/visitors/TypeSignatureTransformer.java

```diff
@@ -329,24 +330,25 @@ protected void visit(TypeSymbol symbol) {
     private String transformExternalRefType(TypeSymbol typeSymbol) {
         // If the module is not anon, imports module.
 
-        String typeName = typeSymbol.name();
+        String typeName = typeSymbol.getName().orElse("");
         if (typeName.isBlank()) {
+            // Name was null or blank
             String typeSignature = typeSymbol.signature();
             typeName = typeSignature.substring(typeSignature.lastIndexOf(':') + 1);
         }
 
-        if (typeSymbol.moduleID().orgName().equals(ANON_MODULE)
-                || (typeSymbol.moduleID().moduleName().equals("lang.annotations")
-                && typeSymbol.moduleID().orgName().equals("ballerina"))) {
+        ModuleID moduleID = typeSymbol.getModule().isPresent() ? typeSymbol.getModule().get().id() : null;
+
+        if (moduleID == null || moduleID.orgName().equals(ANON_MODULE)
+                || (moduleID.moduleName().equals("lang.annotations") && moduleID.orgName().equals("ballerina"))) {
```

Comment:
Usual pattern should be other way; `"lang.annotations".equals(moduleID.moduleName())`